### ISSUE TITLE
Modal hide on escape does not work

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse-frontend#1280]
## Description

Typing `escape` should close modals when `hideOnEsc` is set to true (which it is by default). This currently doesn't work.

## Reproduction

1.  Go to https://wordpress.github.io/openverse-frontend/?path=/story/components-vmodal--default
2.  Open the modal
3.  Type the Escape key on your keyboard
4. The modal should close, instead the disclosure is focused

The problem is that the keydown event is being default prevented, causing the `onKeyDown` handler in the `useDialogContent` composable to early return and never call the `hide` function. I can't find what is causing the default prevention on the event though, it's super mysterious!

## Environment

I've confirmed this on Firefox and Chromium on Linux

## Additional context

Found while working on https://github.com/WordPress/openverse-frontend/pull/1279

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

**Comment by obulat** [comment]
I think this is caused by the `focus-trap` library, which prevents the default on <kbd>Escape</kbd>.
We can stop `focus-trap` from preventing the default by passing `:escape-deactivates="hideOnEsc"` to it. However, we would need to deactivate the `focus-trap` manually, then.
